File: VB/WindowsApplication93/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid

Namespace WindowsApplication93
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			For i As Integer = 0 To 99
				dataTable1.Rows.Add(New Object() {i Mod 5, "Product " & i, i Mod 7 })
			Next i
			gridView1.OptionsView.AllowCellMerge = True
			For i As Integer = 0 To gridView1.Columns.Count - 1
				If gridView1.Columns(i).FieldName = "CategoryID" Then
					gridView1.Columns(i).OptionsColumn.AllowMerge = DevExpress.Utils.DefaultBoolean.True
				Else
					gridView1.Columns(i).OptionsColumn.AllowMerge = DevExpress.Utils.DefaultBoolean.False
				End If
			Next i
		End Sub

		Private Sub gridView1_CellMerge(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Grid.CellMergeEventArgs) Handles gridView1.CellMerge
			Dim view As GridView = TryCast(sender, GridView)
			Try
				If (e.Column.FieldName = "CategoryID") Then
					Dim value1 As Integer = Convert.ToInt32(view.GetRowCellValue(e.RowHandle1, e.Column))
					Dim value2 As Integer = Convert.ToInt32(view.GetRowCellValue(e.RowHandle2, e.Column))

					e.Merge = (value1 = value2)
					e.Handled = True
					Return
				End If
			Catch ex As Exception
			End Try
		End Sub
	End Class
End Namespace

File: CS/WindowsApplication93/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;

namespace WindowsApplication93
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            for (int i = 0; i < 100; i++)
                dataTable1.Rows.Add(new object[] {i %5, "Product " + i, i %7 });
            gridView1.OptionsView.AllowCellMerge = true;
            for (int i = 0; i < gridView1.Columns.Count; i++)
                if (gridView1.Columns[i].FieldName == "CategoryID")
                    gridView1.Columns[i].OptionsColumn.AllowMerge = DevExpress.Utils.DefaultBoolean.True;
                else
                    gridView1.Columns[i].OptionsColumn.AllowMerge =  DevExpress.Utils.DefaultBoolean.False;
        }

        private void gridView1_CellMerge(object sender, DevExpress.XtraGrid.Views.Grid.CellMergeEventArgs e)
        {
            GridView view = sender as GridView;
            try
            {
                if ((e.Column.FieldName == "CategoryID"))
                {
                    int value1 = Convert.ToInt32(view.GetRowCellValue(e.RowHandle1, e.Column));
                    int value2 = Convert.ToInt32(view.GetRowCellValue(e.RowHandle2, e.Column));

                    e.Merge = (value1 == value2);
                    e.Handled = true;
                    return;
                }
            }
            catch (Exception ex)
            {
            }
        }
    }
}